address-space-controller/src/main/java/io/enmasse/controller/ExportsController.java

```diff
@@ -0,0 +1,172 @@
+                        .withUid(addressSpace.getMetadata().getUid())
+                        .build())
+                .endMetadata()
+                .addToData(exportMap)
+                .build();
+
+        ConfigMap existing = client.configMaps().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).get();
+        if (existing != null && !exportMap.equals(existing.getData())) {
+            client.configMaps().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).replace(configMap);
+        } else {
+            client.configMaps().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).createOrReplace(configMap);
+        }
+    }
+
+    private void exportAsService(String name, EndpointStatus endpointStatus, AddressSpace addressSpace) {
+        Service service = new ServiceBuilder()
+                .editOrNewMetadata()
+                .withName(name)
+                .withNamespace(addressSpace.getMetadata().getNamespace())
+                .addToOwnerReferences(new OwnerReferenceBuilder()
+                        .withBlockOwnerDeletion(true)
+                        .withApiVersion(addressSpace.getApiVersion())
+                        .withController(true)
+                        .withKind(addressSpace.getKind())
+                        .withName(addressSpace.getMetadata().getName())
+                        .withUid(addressSpace.getMetadata().getUid())
+                        .build())
+                .endMetadata()
+                .editOrNewSpec()
+                .withType("ExternalName")
+                .withExternalName(endpointStatus.getServiceHost())
+                .withPorts(ServiceHelper.toServicePortList(endpointStatus.getServicePorts()))
+                .endSpec()
+                .build();
+
+        Service existing = client.services().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).get();
+        if (existing != null &&
+                (!endpointStatus.getServiceHost().equals(existing.getSpec().getExternalName())
+                        || !ServiceHelper.toServicePortList(endpointStatus.getServicePorts()).equals(existing.getSpec().getPorts()))) {
+            client.services().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).replace(service);
+        } else {
+            client.services().inNamespace(addressSpace.getMetadata().getNamespace()).withName(name).createOrReplace(service);
+        }
+    }
+
+    private static Map<String, String> buildExportMap(EndpointStatus endpointStatus) {
+        Map<String, String> map = new HashMap<>();
+        map.put("service.host", endpointStatus.getServiceHost());
+        for (Map.Entry<String, Integer> portEntry : endpointStatus.getServicePorts().entrySet()) {
+            map.put("service.port." + portEntry.getKey(), String.valueOf(portEntry.getValue()));
+        }
+        map.put("external.host", endpointStatus.getExternalHost());
```

Comment:
Can't external host be null? If so, should we omit from the map?

Reply:
Done.